### PR TITLE
Python: Fix streaming path to emit mcp_server_tool_result on output_item.done instead of output_item.added

### LINKED ISSUE #4814: Python: [Bug]: Streaming path never delivers mcp_server_tool_result content — parity gap with non-streaming and agent-as-tool paths
### Description

In the streaming path, `mcp_server_tool_result` is emitted immediately on
`response.output_item.added` (`status='calling'`, `output=None`) as a UI lifecycle
signal only. The actual tool output is never forwarded because `response.mcp_call.completed`
has no handler and falls through to the debug-level `case _` branch.

This creates a parity gap across three paths:

| Path | `mcp_server_tool_result` content |
|---|---|
| Non-streaming (`stream=False`) | ✅ Populated from `item.output` |
| Agent-as-tool (local orchestrator) | ✅ Populated via `TOOL_CALL_RESULT` |
| Streaming (`stream=True`) | ❌ Always `output=None` |

UI layers built on the streaming path (e.g. AG-UI/CopilotKit) cannot display MCP tool
outputs to the user, despite this being supported in both the non-streaming path and the
when using an agent as a tool.

---

## Root Cause

In `_parse_chunk_from_openai`, the `response.output_item.added` handler for `mcp_call`
eagerly emits both `mcp_server_tool_call` and `mcp_server_tool_result` at dispatch time,
before the MCP server has responded:

```python
# _parse_chunk_from_openai → response.output_item.added → mcp_call
case "mcp_call":
    call_id = getattr(event_item, "id", None) or getattr(event_item, "call_id", None) or ""
    contents.append(
        Content.from_mcp_server_tool_call(
            call_id=call_id,
            tool_name=getattr(event_item, "name", "") or "",
            server_name=getattr(event_item, "server_label", None),
            arguments=getattr(event_item, "arguments", None),
            raw_representation=event_item,
        )
    )
    result_output = (
        getattr(event_item, "result", None)
        or getattr(event_item, "output", None)   # ← always None at this point
        or getattr(event_item, "outputs", None)
    )
    # ...
    contents.append(
        Content.from_mcp_server_tool_result(
            call_id=call_id,
            output=parsed_output,   # ← always None
            raw_representation=event_item,
        )
    )
```

`response.mcp_call.completed` — which carries the actual output — is listed in the event
type comment block but has no handler, causing it to be silently discarded:

```python
# Line 1766 — listed in comment only, never handled:
# ResponseMcpCallCompletedEvent,
```

The non-streaming path confirms the intent — `item.output` is conditionally populated,
proving the framework authors knew output could exist:

```python
# _parse_response_from_openai — non-streaming, works correctly
case "mcp_call":
    contents.append(Content.from_mcp_server_tool_call(...))
    if item.output is not None:   # ← guard proves output can be populated
        contents.append(
            Content.from_mcp_server_tool_result(
                output=[Content.from_text(text=item.output)],
                ...
            )
        )
```

---

## Steps to Reproduce

1. Configure a Foundry agent with a hosted MCP server tool
2. Call the agent with `stream=True`
3. Observe `mcp_server_tool_result` events with `output=None` in the content stream
4. Confirm `response.mcp_call.completed` events are logged only at DEBUG as "Unparsed event"
5. Call the same agent with `stream=False`
6. Observe `mcp_server_tool_result` events with `output` correctly populated

---

## Expected Behaviour

`mcp_server_tool_result` emitted in the streaming path should carry the tool output,
matching the non-streaming path. The result should be emitted when
`response.mcp_call.completed` fires, not at `response.output_item.added`.

---

## Actual Behaviour

`mcp_server_tool_result` is emitted immediately on `response.output_item.added` with
`output=None`. The `response.mcp_call.completed` event is dropped. The frontend receives
a lifecycle signal but no content, making it impossible to display MCP tool outputs to
the user in streaming mode.

---

## Suggested Fix

Remove the premature `mcp_server_tool_result` emission from the
`response.output_item.added` / `mcp_call` branch, and add a handler for
`response.mcp_call.completed`:

```python
# response.output_item.added / mcp_call — emit call only, defer result
case "mcp_call":
    call_id = getattr(event_item, "id", None) or getattr(event_item, "call_id", None) or ""
    contents.append(
        Content.from_mcp_server_tool_call(
            call_id=call_id,
            tool_name=getattr(event_item, "name", "") or "",
            server_name=getattr(event_item, "server_label", None),
            arguments=getattr(event_item, "arguments", None),
            raw_representation=event_item,
        )
    )
    # Result deferred to response.mcp_call.completed

# New handler — emit result with actual output when MCP call completes
case "response.mcp_call.completed":
    item = event.item
    call_id = getattr(item, "id", None) or ""
    output_text = getattr(item, "output", None)
    parsed_output = (
        [Content.from_text(text=output_text)]
        if isinstance(output_text, str)
        else None
    )
    contents.append(
        Content.from_mcp_server_tool_result(
            call_id=call_id,
            output=parsed_output,
            raw_representation=event,
        )
    )
```

---

## Workaround

Use `stream=False`. The non-streaming path correctly populates `output` at the cost of
losing progressive text rendering

### Code Sample

```markdown

```

### Error Messages / Stack Traces

```markdown

```

### Package Versions

agent-framework-core==1.0.0rc5 agent-framework-azure-ai==1.0.0rc5

### Python Version

Python 3.12

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by giles17**
### Motivation and Context

The streaming code path attempted to extract MCP tool results from `response.output_item.added` events, where the result/output fields are not yet populated. This meant `mcp_server_tool_result` content was either empty or malformed during streaming, breaking parity with the non-streaming and agent-as-tool paths.

Fixes #4814

### Description

The root cause was that `_parse_chunk_from_openai` tried to read `result`/`output`/`outputs` from the MCP call item at the `response.output_item.added` stage, before the API has populated those fields. The fix removes that premature extraction and instead handles MCP call results in a new `response.output_item.done` case, where the completed item carries the final `output` string. This aligns streaming behavior with the non-streaming path and correctly emits `Content.from_mcp_server_tool_result` with the actual tool output.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [Contribution Guidelines](https://github.com/microsoft/agent-framework/blob/main/CONTRIBUTING.md)
- [X] All unit tests pass, and I have added new tests where possible
- [ ] **Is this a breaking change?** If yes, add "[BREAKING]" prefix to the title of the PR.

---

> Note: PR autogenerated by giles17's agent

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>packages/openai/agent_framework_openai</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py">_chat_client.py</a></td><td>887</td><td>133</td><td>85%</td><td><a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L346">346</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L348-L349">348&ndash;349</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L406-L407">406&ndash;407</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L409">409</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L441">441</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L500-L503">500&ndash;503</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L507-L508">507&ndash;508</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L513-L514">513&ndash;514</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L524-L525">524&ndash;525</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L532">532</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L547-L553">547&ndash;553</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L574">574</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L582">582</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L605">605</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L702">702</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L801">801</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L860">860</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L862">862</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L864">864</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L866">866</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L932">932</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L946">946</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1026">1026</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1036">1036</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1041">1041</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1084">1084</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1163">1163</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1180">1180</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1193">1193</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1258">1258</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1351">1351</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1356">1356</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1360-L1362">1360&ndash;1362</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1366-L1367">1366&ndash;1367</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1433">1433</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1462">1462</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1468">1468</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1478">1478</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1484">1484</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1489">1489</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1495">1495</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1500-L1501">1500&ndash;1501</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1562">1562</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1584-L1585">1584&ndash;1585</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1600-L1601">1600&ndash;1601</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1619-L1620">1619&ndash;1620</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1661-L1664">1661&ndash;1664</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1826">1826</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1881">1881</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1883">1883</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L1963-L1971">1963&ndash;1971</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2093">2093</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2128">2128</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2143">2143</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2163-L2173">2163&ndash;2173</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2186">2186</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2197-L2201">2197&ndash;2201</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2215">2215</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2229-L2240">2229&ndash;2240</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2249">2249</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2281-L2284">2281&ndash;2284</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2292-L2293">2292&ndash;2293</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2295-L2297">2295&ndash;2297</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2311-L2313">2311&ndash;2313</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2323-L2324">2323&ndash;2324</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2330">2330</a>, <a href="https://github.com/microsoft/agent-framework/blob/daf4cb5768fcecc89af58f39944d2c6b7ea87546/packages/openai/agent_framework_openai/_chat_client.py#L2345">2345</a></td></tr><tr><td><b>TOTAL</b></td><td><b>27963</b></td><td><b>3412</b></td><td><b>87%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 5421 | 20 :zzz: | 0 :x: | 0 :fire: | 1m 26s :stopwatch: |

File: python/packages/openai/tests/openai/test_openai_chat_client.py [hunk 1]
```diff
@@ -1209,18 +1212,97 @@ def test_parse_chunk_from_openai_with_mcp_call_result() -> None:
...
+
+    assert len(update.contents) == 1
+    result_content = update.contents[0]
+
     assert result_content.type == "mcp_server_tool_result"
-    assert result_content.call_id in ["mcp_call_456", "call_456"]
-    # Verify the output was parsed
+    assert result_content.call_id == "mcp_fallback_123"
     assert result_content.output is not None
+    assert result_content.output[0].text == "fallback result"
```
Comment (giles17) [inline python/packages/openai/tests/openai/test_openai_chat_client.py:1305]: This test is missing an assertion on `raw_representation`, unlike the two sibling tests which now assert `result_content.raw_representation is mock_item`. Since the `raw_representation=done_item` fix is part of this PR, add the same assertion here for consistency and full validation.

```suggestion
    assert result_content.call_id == "mcp_fallback_123"
    assert result_content.output is not None
    assert result_content.output[0].text == "fallback result"
    assert result_content.raw_representation is mock_item
```

**Review by giles17 — COMMENTED** [review]
## Automated Code Review

**Reviewers:** 4 | **Confidence:** 93%

### ✓ Correctness
> This PR makes two small, correct fixes to the streaming MCP call handling in `_parse_chunk_from_openai`: (1) adds a `call_id` attribute fallback when extracting the call identifier from a done mcp_call item, and (2) fixes `raw_representation` to pass the item object instead of the event wrapper, aligning with the non-streaming path and the `response.output_item.added` streaming path. Tests are well-constructed and correctly validate the changes. No issues found.

### ✓ Security Reliability
> The changes are minor, correct, and improve reliability: (1) adding a `call_id` attribute fallback when `id` is missing on the `done_item` guards against different event object shapes from the OpenAI API, and (2) passing `done_item` instead of `event` as `raw_representation` is consistent with the non-streaming path (line 1560) and the streaming `output_item.added` handler (line 1932), which both pass the item object. The test coverage for the fallback and raw_representation assertions is adequate. No security or reliability issues found.

### ✗ Test Coverage
> The diff fixes two bugs in the streaming MCP call handler: (1) adds a `call_id` fallback when `id` is missing, and (2) passes `done_item` instead of `event` as `raw_representation`. Tests cover the happy path, null-output, and call_id fallback cases, plus verify `raw_representation`. However, the new fallback test is missing a `raw_representation` assertion, and there is no test for the case where both `id` and `call_id` are absent (empty-string fallback). The broader issue from the linked discussion—missing handler for `response.mcp_call.completed`—is not addressed here, but that is a separate feature/fix beyond this PR's scope.

### ✓ Design Approach
> The two changes are well-motivated and consistent with the rest of the codebase. The `call_id` fallback in `response.output_item.done` now mirrors the identical pattern already used in `response.output_item.added` (line 1925), and `raw_representation=done_item` aligns with both the non-streaming path (line 1560) and the streaming `mcp_call` added handler (line 1932), which all use the item rather than the outer event wrapper. No design-level concerns found. The only gap is a missing `raw_representation` assertion in the new fallback test.

### Suggestions
- Add `assert result_content.raw_representation is mock_item` to `test_parse_chunk_from_openai_with_mcp_output_item_done_call_id_fallback` for consistency with the other two MCP done tests, since the `raw_representation=done_item` fix is part of what this PR explicitly changes.
- The non-streaming path at line 1545 uses `item.id` directly without the `call_id` fallback. Consider applying the same defensive fallback (`getattr(item, 'id', None) or getattr(item, 'call_id', None) or ''`) there for consistency, so both paths handle unexpected API shapes gracefully.
- Add a test case where the mock item has neither `id` nor `call_id` to verify the empty-string fallback path (the third `or ""` branch).

---
*Automated review by giles17's agents*

File: python/packages/openai/agent_framework_openai/_chat_client.py [hunk 2]
```diff
@@ -2222,6 +2202,21 @@ def _get_ann_value(key: str) -> Any:
...
+                    output_text = getattr(done_item, "output", None)
+                    parsed_output: list[Content] | None = (
+                        [Content.from_text(text=output_text)] if isinstance(output_text, str) else None
+                    )
+                    contents.append(
+                        Content.from_mcp_server_tool_result(
+                            call_id=call_id,
+                            output=parsed_output,
+                            raw_representation=event,
+                        )
```
[the remark this answers is not in the document]
Comment (giles17) [reply python/packages/openai/agent_framework_openai/_chat_client.py:2404]: Fixed in `45f5b614`.

File: python/packages/openai/tests/openai/test_openai_chat_client.py (hunk 1 above)
Comment (giles17) [reply python/packages/openai/tests/openai/test_openai_chat_client.py:1305]: Fixed in `45f5b614`.

File: python/packages/openai/tests/openai/test_openai_chat_client.py [hunk 3]
```diff
@@ -1209,18 +1212,124 @@ def test_parse_chunk_from_openai_with_mcp_call_result() -> None:
...
+    result_content = update.contents[0]
+
+    assert result_content.type == "mcp_server_tool_result"
+    assert result_content.call_id == "mcp_fallback_123"
+    assert result_content.output is not None
+    assert result_content.output[0].text == "fallback result"
+    assert result_content.raw_representation is mock_item
+
+
+def test_parse_chunk_from_openai_with_mcp_output_item_done_no_id_fallback() -> None:
```
Comment (giles17) [inline python/packages/openai/tests/openai/test_openai_chat_client.py:1309]: This test exercises the **streaming** path (`_parse_chunk_from_openai`, line 2208), which already had the `getattr` fallback before this PR. The actual code change is at line 1545 in `_parse_response_from_openai`. Add analogous non-streaming tests using `MagicMock(spec=[])` items (so `.id` isn't auto-created) passed to `_parse_response_from_openai` to directly cover the changed line.

File: python/packages/openai/agent_framework_openai/_chat_client.py [hunk 4]
```diff
@@ -1542,7 +1542,7 @@ def _parse_response_from_openai(
                         )
                     )
                 case "mcp_call":
-                    call_id = item.id
+                    call_id = getattr(item, "id", None) or getattr(item, "call_id", None) or ""
```
Comment (giles17) [inline python/packages/openai/agent_framework_openai/_chat_client.py:1731]: This is the actual code change—the non-streaming fallback logic—but no test exercises this line with a missing `id` attribute. The existing test at line 1161 uses `MagicMock()` without `spec`, which auto-creates `.id` and never tests the fallback. The new tests all call `_parse_chunk_from_openai`, not `_parse_response_from_openai`.

File: python/packages/openai/tests/openai/test_openai_chat_client.py [hunk 5]
```diff
@@ -1209,18 +1212,124 @@ def test_parse_chunk_from_openai_with_mcp_call_result() -> None:
...
+
+    update = client._parse_chunk_from_openai(mock_event, options={}, function_call_ids=function_call_ids)
+
+    assert len(update.contents) == 1
+    result_content = update.contents[0]
+
     assert result_content.type == "mcp_server_tool_result"
-    assert result_content.call_id in ["mcp_call_456", "call_456"]
-    # Verify the output was parsed
+    assert result_content.call_id == ""
```
Comment (giles17) [inline python/packages/openai/tests/openai/test_openai_chat_client.py:1329]: Asserting `call_id == ""` codifies silent data corruption: if two MCP calls both lack an `id`, they become indistinguishable at the consumer layer. Consider asserting that a warning was logged (or that an exception is raised) rather than baking in the broken-correlation path as a passing test.

```suggestion
    assert result_content.call_id == ""  # TODO: should warn/raise instead of silently producing an empty call_id
```

File: python/packages/openai/tests/openai/test_openai_chat_client.py (hunk 3 above)
Comment (giles17) [reply python/packages/openai/tests/openai/test_openai_chat_client.py:1309]: Fixed in `05341221`.

File: python/packages/openai/agent_framework_openai/_chat_client.py (hunk 4 above)
Comment (giles17) [reply python/packages/openai/agent_framework_openai/_chat_client.py:1731]: Fixed in `05341221`.

File: python/packages/openai/tests/openai/test_openai_chat_client.py (hunk 5 above)
Comment (giles17) [reply python/packages/openai/tests/openai/test_openai_chat_client.py:1329]: Fixed in `05341221`.
